scripts/ci/pre_commit/pre_commit_breeze_generate_autocomplete_scripts.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python3
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import os
+import sys
+from os.path import abspath, dirname, join
+
+AIRFLOW_SOURCES_DIR = abspath(join(dirname(__file__), os.pardir, os.pardir, os.pardir))
+
+sys.path.insert(0, AIRFLOW_SOURCES_DIR)
+# flake8: noqa: F401
+
+
+def generate_autocompletion_script(shell_type: list):
+    # This command will generate the autocomplete script in the given path
+    # https://click.palletsprojects.com/en/7.x/bashcomplete/#activation-script
+    # _FOO_BAR_COMPLETE=source_bash foo-bar > foo-bar-complete.sh
+
+    for shell in shell_type:
+        autocomplete_path = os.path.join(
+            AIRFLOW_SOURCES_DIR, "dev/breeze/autocomplete", f"Breeze2-complete.{shell}"
+        )
+        os.system(f"_BREEZE2_COMPLETE=source_{shell} Breeze2 > {autocomplete_path}")
```

Comment:
It woudl be great to rewrite it using subprocess.run. It is much more portable - see an example here: https://github.com/apache/airflow/pull/19992/files#diff-5bb3ebfd8f8fb007b4b1029ede846d6215c2467929b515f8ea05720f82ef4fc9R72 - when you specify shell=True, it should find `Breeze` command on the PATH as well.